File: Modules/Media/Utilities/RegExUtility.cs
//
// DotNetNuke® - http://www.dotnetnuke.com
// Copyright (c) 2002-2013
// by DotNetNuke Corporation
//
// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated 
// documentation files (the "Software"), to deal in the Software without restriction, including without limitation 
// the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and 
// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
//
// The above copyright notice and this permission notice shall be included in all copies or substantial portions 
// of the Software.
//
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED 
// TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL 
// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF 
// CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
// DEALINGS IN THE SOFTWARE.
//

//INSTANT C# NOTE: Formerly VB project-level imports:
using DotNetNuke;
using DotNetNuke.Common;
using DotNetNuke.Common.Utilities;
using DotNetNuke.Services.Exceptions;
using DotNetNuke.Services.Localization;
using System;
using System.Collections;
using System.Data;
using System.Diagnostics;

using System.Text.RegularExpressions;

namespace DotNetNuke.Modules.Media
{

	public sealed class RegExUtility
	{

#region  Constants 

		private const string POSITIVE_ONLY_PATTERN = "^\\d+(\\.\\d+)*?$";
		private const string NEGATIVE_ALLOWED_PATTERN = "^\\-*\\d+(\\.\\d+)*?$";
		private const string BOOLEAN_PATTERN = "^(1|0|true|false)$";

#endregion

		/// <summary>
		/// IsNumber - this method uses a regular expression to determine if the value object is in a valid numeric format.
		/// </summary>
		/// <param name="Value">Object - the object to parse to see if it's a number</param>
		/// <returns>If true, the Value object was in a valid numeric format</returns>
		/// <remarks>
		/// This method does not consider commas (,) to be a valid character. This overload defaults PositiveOnly to True.
		/// </remarks>
		/// <history>
		/// [wstrohl] - 20100130 - created
		/// </history>
		public static bool IsNumber(object Value)
		{
			return IsNumber(Value, true);
		}

		/// <summary>
		/// IsNumber - this method uses a regular expression to determine if the value object is in a valid numeric format.
		/// </summary>
		/// <param name="Value">Object - the object to parse to see if it's a number</param>
		/// <param name="PositiveOnly">Boolean - if true, a negative number will be considered valid</param>
		/// <returns>If true, the Value object was in a valid numeric format</returns>
		/// <remarks>
		/// This method does not consider commas (,) to be a valid character.
		/// </remarks>
		/// <history>
		/// [wstrohl] - 20100130 - created
		/// </history>
		public static bool IsNumber(object Value, bool PositiveOnly)
		{

			if (Value == null)
			{
				return false;
			}

			if (PositiveOnly)
			{
				return Regex.IsMatch(Value.ToString(), POSITIVE_ONLY_PATTERN);
			}
			else
			{
				return Regex.IsMatch(Value.ToString(), NEGATIVE_ALLOWED_PATTERN);
			}

		}

		/// <summary>
		/// IsBoolean - this method uses a regular expression to determine if the value object is in a valid boolean format.
		/// </summary>
		/// <param name="Value">Object - the object to parse to see if it is in a boolean fomat</param>
		/// <returns>If true, the Value object was in a valid boolean format</returns>
		/// <remarks>
		/// This method looks for one of the following: 1, 0, true, false (case insensitive)
		/// </remarks>
		/// <history>
		/// [wstrohl] - 20100130 - created
		/// </history>
		public static bool IsBoolean(object Value)
		{

			if (Value == null)
			{
				return false;
			}

			return Regex.IsMatch(Value.ToString(), BOOLEAN_PATTERN, RegexOptions.IgnoreCase);

		}

	}

}

File: Modules/WillStrohl.ContactCollector/Components/RegExUtility.vb
'
' Will Strohl (will.strohl@gmail.com)
' http://www.willstrohl.com
'
'Copyright (c) 2015, Will Strohl
'All rights reserved.
'
'Redistribution and use in source and binary forms, with or without modification, are 
'permitted provided that the following conditions are met:
'
'Redistributions of source code must retain the above copyright notice, this list of 
'conditions and the following disclaimer.
'
'Redistributions in binary form must reproduce the above copyright notice, this list 
'of conditions and the following disclaimer in the documentation and/or other 
'materials provided with the distribution.
'
'Neither the name of Will Strohl, Contact Collector, nor the names of its contributors may be 
'used to endorse or promote products derived from this software without specific prior 
'written permission.
'
'THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY 
'EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES 
'OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT 
'SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, 
'INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED 
'TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR 
'BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
'CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
'ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH 
'DAMAGE.
'

Imports System.Text.RegularExpressions

Namespace WillStrohl.Modules.ContactCollector

    Public NotInheritable Class RegExUtility

#Region " Constants "

        Private Const POSITIVE_ONLY_PATTERN As String = "^\d+(\.\d+)*?$"
        Private Const NEGATIVE_ALLOWED_PATTERN As String = "^\-*\d+(\.\d+)*?$"
        Private Const BOOLEAN_PATTERN As String = "^(1|0|true|false)$"

#End Region

        ''' <summary>
        ''' IsNumber - this method uses a regular expression to determine if the value object is in a valid numeric format.
        ''' </summary>
        ''' <param name="Value">Object - the object to parse to see if it's a number</param>
        ''' <returns>If true, the Value object was in a valid numeric format</returns>
        ''' <remarks>
        ''' This method does not consider commas (,) to be a valid character. This overload defaults PositiveOnly to True.
        ''' </remarks>
        ''' <history>
        ''' [wstrohl] - 20100130 - created
        ''' </history>
        Public Overloads Shared Function IsNumber(ByVal Value As Object) As Boolean
            Return IsNumber(Value, True)
        End Function

        ''' <summary>
        ''' IsNumber - this method uses a regular expression to determine if the value object is in a valid numeric format.
        ''' </summary>
        ''' <param name="Value">Object - the object to parse to see if it's a number</param>
        ''' <param name="PositiveOnly">Boolean - if true, a negative number will be considered valid</param>
        ''' <returns>If true, the Value object was in a valid numeric format</returns>
        ''' <remarks>
        ''' This method does not consider commas (,) to be a valid character.
        ''' </remarks>
        ''' <history>
        ''' [wstrohl] - 20100130 - created
        ''' </history>
        Public Overloads Shared Function IsNumber(ByVal Value As Object, ByVal PositiveOnly As Boolean) As Boolean

            If Value Is Nothing Then
                Return False
            End If

            If PositiveOnly Then
                Return Regex.IsMatch(Value.ToString, POSITIVE_ONLY_PATTERN)
            Else
                Return Regex.IsMatch(Value.ToString, NEGATIVE_ALLOWED_PATTERN)
            End If

        End Function

        ''' <summary>
        ''' IsBoolean - this method uses a regular expression to determine if the value object is in a valid boolean format.
        ''' </summary>
        ''' <param name="Value">Object - the object to parse to see if it is in a boolean fomat</param>
        ''' <returns>If true, the Value object was in a valid boolean format</returns>
        ''' <remarks>
        ''' This method looks for one of the following: 1, 0, true, false (case insensitive)
        ''' </remarks>
        ''' <history>
        ''' [wstrohl] - 20100130 - created
        ''' </history>
        Public Overloads Shared Function IsBoolean(ByVal Value As Object) As Boolean

            If Value Is Nothing Then
                Return False
            End If

            Return Regex.IsMatch(Value.ToString, BOOLEAN_PATTERN, RegexOptions.IgnoreCase)

        End Function

    End Class

End Namespace